src/views/AnnualReport.vue

```diff
@@ -803,13 +809,10 @@ export default class AnnualReport extends Mixins(
   }
 
   onClickSaveFinish (): void {
-    const filingId = +this.savedFiling?.header?.filingId
     // safety check
-    if (filingId > 0) {
+    if (this.filingId > 0) {
       // changes were saved, so clear flag
       this.haveChanges = false
-      // save filing ID for future updates
-      this.filingId = filingId
```

Comment:
In the old code, `this.filingId` was only updated when the "finish" part of the step ran.

The bug fix is that we want to save the new filingId even if we get a save/pay error, so that we can retry using the new filingId (which becomes a PUT instead of a POST).